mlflow/transformers.py

```diff
@@ -0,0 +1,799 @@
+                        this behavior, supply a valid ``task`` argument during model logging or
+                        saving. Default is "pipeline".
+    :param kwargs: Optional configuration options for loading of a ``transformers`` object.
+                   For information on parameters and their usage, see
+                   `transformers documentation <https://huggingface.co/docs/transformers/index>`_.
+    :return: A ``transformers`` model instance or a dictionary of components
+    """
+
+    import transformers
+
+    model_uri = str(model_uri)
+
+    local_model_path = _download_artifact_from_uri(artifact_uri=model_uri, output_path=dst_path)
+
+    flavor_config = _get_flavor_configuration_from_uri(model_uri, FLAVOR_NAME, _logger)
+
+    if return_type == "pipeline" and flavor_config.get(_PROCESSOR_TYPE_KEY, None):
+        raise MlflowException(
+            "This model has been saved with a processor. Processor objects are "
+            "not compatible with Pipelines. Please load this model by specifying "
+            "the 'return_type'='components'."
+        )
+
+    _add_code_from_conf_to_system_path(local_model_path, flavor_config)
+
+    components = _load_model(local_model_path, flavor_config)
+    if return_type == "pipeline":
+        components.update(**kwargs)
+        return transformers.pipeline(**components)
+    elif return_type == "components":
+        return components
+    else:
+        raise MlflowException(
+            f"The specified return_type mode '{return_type}' is unsupported. "
+            "Please select one of: 'pipeline' or 'components'."
+        )
+
+
+def _load_model(path: str, flavor_config):
+    """
+    Loads components from a locally serialized ``Pipeline`` object.
+    """
+    import transformers
+
+    local_path = pathlib.Path(path)
+    pipeline_path = local_path.joinpath(
+        flavor_config.get(_PIPELINE_BINARY_KEY, _PIPELINE_BINARY_FILE_NAME)
+    )
+
+    model_instance = getattr(transformers, flavor_config[_PIPELINE_MODEL_TYPE_KEY])
+    conf = {
+        "task": flavor_config[_TASK_KEY],
+        "model": model_instance.from_pretrained(pipeline_path),
+    }
+
+    if flavor_config.get(_PROCESSOR_TYPE_KEY, None):
+        conf[_PROCESSOR_KEY] = _load_component(
+            local_path, _PROCESSOR_KEY, flavor_config[_PROCESSOR_TYPE_KEY]
+        )
+
+    for component_key in flavor_config[_COMPONENTS_BINARY_KEY]:
+        component_type_key = f"{component_key}_type"
+        component_type = flavor_config[component_type_key]
+        conf[component_key] = _load_component(local_path, component_key, component_type)
+    return conf
+
+
+def _hub_access():
+    """
+    Wrapper around importing huggingface_hub for ModelCard retrieval, providing access to the
+    module if the library is installed, else noop.
+    """
+    import importlib
+
+    try:
+        return importlib.import_module("huggingface_hub")
+    except ImportError:
+        pass
```

Comment:
Can we use `contextlib.suppress` (which is shorter)?

Reply:
changed! good call-out :)